src/core/reactor.cc

```diff
@@ -4584,7 +4596,16 @@ void smp::configure(const smp_options& smp_opts, const reactor_options& reactor_
     _qs_owner = decltype(smp::_qs_owner){new smp_message_queue* [smp::count], qs_deleter{}};
     _qs = _qs_owner.get();
     for(unsigned i = 0; i < smp::count; i++) {
-        smp::_qs_owner[i] = reinterpret_cast<smp_message_queue*>(operator new[] (sizeof(smp_message_queue) * smp::count));
+        smp::_qs_owner[i] = reinterpret_cast<smp_message_queue*>(operator new[] (sizeof(smp_message_queue) * smp::count
+        // smp_message_queue has members with hefty alignment requirements.
+        // if we are reactor thread, or not running with dpdk, doing this
+        // new default aligned seemingly works, as does reordering
+        // dlinit dependencies (ugh). But we should enforce calling out to
+        // aligned_alloc, instead of pure malloc, if possible.
+#ifdef __cpp_aligned_new
+            , std::align_val_t(alignof(smp_message_queue))
+#endif
+        ));
```

Comment:
__cpp_aligned_new is from 2018 (gcc and clang), we can avoid the #ifdef

Reply:
Just being consistent with the definition of the aligned alloc functions in memory.cc.

Reply:
probably we can drop them all together in another change?

Reply:
FWIW, Avi is removing them in #2150 .

Reply:
Ok, removed.